FUZZ-5131: Add claude.md file (#41)
ctrliq/ciq-fuzzball-catalog: CLAUDE.md

# CLAUDE.md

This file provides guidance to Claude Code (claude.ai/code) when working with code in this repository.

## Overview

This repository defines applications for use with Fuzzball (see the [Fuzzball
documentation](https://ui.stable.fuzzball.ciq.dev/docs/). Applications are
defined in the `applications` directory. Each application is housed in a subdirectory
of `applications`. The required files for each application are

- `metadata.md`: Contains structured information in the front matter and are
  otherwise a free form description of the application.
- `template.yaml`: Fuzzball workflow files as described in
  https://ui.stable.fuzzball.ciq.dev/docs/appendices/workflow-syntax/ and contain
  golang template strings augmented with functions from the
  [slim-sprig](https://github.com/go-task/slim-sprig) library.
- `values.yaml`: Describe values used to template the `template.yaml` files.

> located at the top level of the directory. Other (arbitrary, optional) files may

Reviewer: ```suggestion
These files must be located at the top level of the directory. Other (arbitrary, optional) files may
```

be included as well.

> Running
> 
> ```sh
> fuzzball application render-template <INPUT_FILE> <VALUES_FILE>
> ```
> 
> with a `template.yaml` and a `values.yaml` file should yield a valid Fuzzfile workflow.

Reviewer: ```suggestion
Running the following command with a `template.yaml` and a `values.yaml` file should yield a valid workflow template.

```sh
fuzzball application render-template <INPUT_FILE> <VALUES_FILE>
```

```

## Style guide

> - Application templates should follow a style consistent with existing templates

Reviewer: ```suggestion
- Application templates should follow a style consistent with existing templates.
```
I wonder if we should instead point to a subset of archetypal templates.  If we leave this as the guidance, will the style drift over time as Claude accepts new templates as part of the existing style?

Author: Hm. That's a good point. How about the following 5 which fit the existing style guide pretty well:
- bwa_alignment
- specfrm3d
- blast
- gromacs_gpu
- openfoam_motorbike

Reviewer: That sounds good to me.
